components/carousel/README.md

```diff
@@ -0,0 +1,115 @@
+
+~~~~{.html}
+<script src="./node_modules/coherent-gameface-carousel/dist/carousel.production.min.js"></script>
+~~~~
+
+* add the coherent-gameface-carousel component to your html:
+
+~~~~{.html}
+<gameface-carousel>
+    <div slot="carousel-content">0</div>
+    <div slot="carousel-content">1</div>
+    <div slot="carousel-content">2</div>
+    <div slot="carousel-content">3</div>
+    <div slot="carousel-content">4</div>
+    <div slot="carousel-content">5</div>
+    <div slot="carousel-content">6</div>
+    <div slot="carousel-content">7</div>
+</gameface-carousel>
+~~~~
+
+This is all! Load the file in Gameface to see the coherent-gameface-carousel.
+
+## Usage with JavaScript:
+
+If you wish to import the modules using JavaScript you can remove the script tags
+which import the components and the coherent-gameface-carousel from the node_modules folder and import them like this:
+
+~~~~{.js}
+import { Carousel } from 'coherent-gameface-carousel';
+~~~~
+
+or simply
+
+~~~~{.js}
+import 'coherent-gameface-carousel';
+~~~~
+
+Note that this approach requires a module bundler like [Webpack](https://webpack.js.org/) or [Rollup](https://rollupjs.org/guide/en/) to resolve the
+modules from the node_modules folder.
+
+
+Public API
+===================
+
+# Available Slots
+### carousel-item
+
+The carousel supports a dynamic slot that is supposed to hold its items - `carousel-content`. To add an item into this slot specify its slot attribute:
+
+~~~~{.html}
+<div slot="carousel-content">7</div>
+~~~~
+
+All elements that have this slot attribute ill be initially added to the carousel.
+
+### carousel-next and carousel-previous
+
+These slots are reserved for the navigation controls that change the currently active carousel items. By default they are right and left pointing arrows. Set the slot attribute to `next` or `previous` to an element to use custom navigation controls:
+
+~~~~{.html}
+<gameface-carousel>
+    <button slot="carousel-next">Next</div>
+    <button slot="carousel-previous">Prev</div>
+<gameface-carousel>
+~~~~
+
+# Adding items to the carousel
+`addItem(element, index)`
+
+Use the `addItem` method to add a new item:
+
+## Parameters
+
+- `element` - the new carousel item that should be added
+- `[index]` - the index at which to add the new item
+
+~~~~{.js}
+const carouselItem = document.createElement('div');
+carouselItem.textContent = 'Hello, there!';
+
+document.querySelector('gameface-carousel').addItem(cItem);
+~~~~
+
+# Removing items from the carousel
+
+`removeItem(index)`
+
+Use the `removeItem` method to remove an item:
+
+## Parameters
+
+- `[index]` - the index at which to add the new item, if omitted - will remove the first element
+
+~~~~{.js}
+document.querySelector('gameface-carousel').removeItem();
+~~~~
+
+# Updating the page size
+
+Use the `pageSize` setter to change the number of simultaneously visible elements.
+
+~~~~{.js}
+document.querySelector('gameface-carousel').pageSize = 3;
+~~~~
```

Comment:
Remember to run the generation/transferring of the documentation command after changes.